.github/workflows/My-ci-pipeline.yml

```diff
@@ -54,21 +51,35 @@ jobs:
     - name: Publish to Docker hub
       run: docker push wishy78/todo-app:latest
   
-
   cd:
     #if: ${{ github.ref == 'refs/heads/master' && github.event_name == 'pull_request' }}
     #if: ${{ github.ref == 'refs/heads/module10' && github.event_name == 'push' }}
     name: Build and Deploy
     runs-on: ubuntu-latest
     needs: [Publish]
+    env:
+        ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+        ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+        ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+        ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+      
+        TF_VAR_CLIENTID: ${{ secrets.TF_VAR_CLIENTID }}
+        TF_VAR_CLIENTSECRET: ${{ secrets.TF_VAR_CLIENTSECRET }}
+        TF_VAR_DB_NAME: ${{ secrets.TF_VAR_DB_NAME }}
+        TF_VAR_COLLECTION_NAME: ${{ secrets.TF_VAR_COLLECTION_NAME }}
+        TF_VAR_SECRET_KEY: ${{ secrets.TF_VAR_SECRET_KEY }}
+        
     steps:
-    - uses: actions/checkout@v2
-    
-    - uses: satackey/action-docker-layer-caching@v0.0.11
-      continue-on-error: true
-    
+    - uses: actions/checkout@v3
+   
+    - name: initiate Terraform
+      run: terraform init 
+
```

Comment:
I think this works quite nicely, but on your comment re separating the pipelines, if we did want to do that we _could_ do so simply by re-initialising Terraform in the next stage, and then (without having to re-apply anything) it should have access to the latest outputs

Alternatively we could do something with [job outputs](https://stackoverflow.com/questions/59175332/using-output-from-a-previous-job-in-a-new-one-in-a-github-action) I believe, but they look fairly fiddly!

Reply:
yes they are fidderly as spent 4 days on different methods before doing this one, thanks for the advice